src/twisted/conch/ssh/filetransfer.py

```diff
@@ -531,9 +534,16 @@ def openFile(self, filename, flags, attrs):
         """
         data = NS(filename) + struct.pack('!L', flags) + self._packAttributes(attrs)
         d = self._sendRequest(FXP_OPEN, data)
-        self.wasAFile[d] = (1, filename) # HACK
+        d.addCallback(self._cbOpenHandle, ClientFile, filename)
         return d
 
+
+    def _cbOpenHandle(self, handle, wrapperClass, name):
```

Comment:
even if this is private, it needs docstring.
I know that existing code might have fewer docstring but we should try to raise the bar, at least for the new code.

Is `wrapperClass` the right name. Wrapper for what?
They look to me like normal objects which are specialized to handle file or directory requests and that they are not wrapping any other object.

I hope that with the docstring, these things will be clear.

Reply:
I don't know what to say. If we go on this line we can say that the any class in conch.ssh is a wrapper, as it just provides a nice interface for the SSH raw bytes... so FileTransferServer is just a wrapper around the SFTP subsystem.

Maybe call it just `handlingClass` with a docstring along these lines:
'Class to further process the open file or open directory requests.`

In Twisted, a wrapper is usually an object which wraps another object and provides the same interface.

Here we have 2 classes with different API

Reply:
How about `handleClass`? That would make a lot of sense to me: it's the class that is given back to user code to act as a file handle, and it also contains the filetransfer-protocol file handle value. Unless that sounds like a bad name to you I'll change it and add a docstring.

(I agree `wrapperClass` isn't an especially good name, it's just the best name I could think of. It seems like a "wrapper" to me because it mostly passes methods through to the `FileTransferClient`.)

Reply:
+1 for handleClass 

The docstring is needed anyway :)